src/modules/helpers/index.ts

```diff
@@ -4,6 +4,74 @@ import { luhnCheckValue } from './luhn-check';
 import type { RecordKey } from './unique';
 import * as uniqueExec from './unique';
 
+/**
+ * Returns a number based on given RegEx-based quantifier symbol or quantifier values.
+ *
+ * @param faker Faker instance
+ * @param quantifierSymbol Quantifier symbols can be either of these: `?`, `*`, `+`.
+ * @param quantifierMin Quantifier minimum value. If given without a maximum, this will be used as the quantifier value.
+ * @param quantifierMax Quantifier maximum value. Will randomly get a value between the minimum and maximum if both are provided.
+ *
+ * @returns a random number based on the given quantifier parameters.
+ *
+ * @example
+ * getRepetitionsBasedOnQuantifierParameters(this.faker, '*', null, null) // 3
+ * getRepetitionsBasedOnQuantifierParameters(this.faker, null, 10, null) // 10
+ * getRepetitionsBasedOnQuantifierParameters(this.faker, null, 5, 8) // 6
+ *
+ * @since 8.0.0
+ */
+function getRepetitionsBasedOnQuantifierParameters(
+  faker: Faker,
+  quantifierSymbol: string,
+  quantifierMin: string,
+  quantifierMax: string
+) {
+  let repetitions = 0;
+  if (quantifierSymbol) {
+    switch (quantifierSymbol) {
+      case '?': {
+        repetitions = faker.datatype.boolean() ? 0 : 1;
+        break;
+      }
+
+      case '*': {
+        let limit = 1;
+        while (faker.datatype.boolean()) {
+          limit *= 2;
+        }
+
+        repetitions = faker.number.int({ min: 0, max: limit });
+        break;
+      }
+
+      case '+': {
+        let limit = 1;
+        while (faker.datatype.boolean()) {
+          limit *= 2;
+        }
+
+        repetitions = faker.number.int({ min: 1, max: limit });
+        break;
+      }
+
+      default:
+        throw new FakerError('Unknown quantifier symbol provided.');
+    }
+  } else if (quantifierMin && quantifierMax) {
```

Comment:
Is e.g. `quantifierMin == 0` allowed here?

Reply:
I would assume yes because `faker.number.int` will handle the error checking if the user makes the `quantifierMin` more than the `quantiferMax` or other problems related to it.

Reply:
Do we then have to use 

```suggestion
  } else if (quantifierMin != null && quantifierMax != null) {
```

instead?

Reply:
Oh yeah that makes sense since `(quantifierMin && quantifierMax)` will be false if both of them are 0 which is wrong. I will update this soon.